bin/translate_images.py

```diff
@@ -0,0 +1,133 @@
+    """
+    return Translate(
+        destName=out_file,
+        srcDS=ds,
+        bandList=[float(1)],
+        format="NITF",
+        outputType=gdalconst.GDT_Int16,
+        creationOptions=["IC=C8", "PROFILE=NPJE_NUMERICALLY_LOSSLESS"],
+    )
+
+
+def translate_to_wibd(out_file: str, ds: gdal.Dataset) -> gdal.Dataset:
+    """
+    Converts a GeoTIFF dataset to a NITF file in the WIBD format.
+
+    :param out_file: The path to the output NITF file to be created.
+    :param ds: The input GeoTIFF dataset to be converted.
+    :return: The converted NITF dataset.
+    """
+    return Translate(
+        destName=out_file,
+        srcDS=ds,
+        bandList=[1],
+        format="NITF",
+        outputType=gdalconst.GDT_Byte,
+        creationOptions=["IC=C8", "PROFILE=NPJE_NUMERICALLY_LOSSLESS"],
+    )
+
+
+def convert_geotiff_to_nitf(input_dir: str, target_format: TargetFormat, output_dir: Optional[str] = None) -> None:
+    """
+    Convert GeoTIFF files in the input directory to NITF format (either greyscale or WIBD).
+
+    :param input_dir: The directory containing the input GeoTIFF files.
+    :param target_format: The target format for conversion, either GREYSCALE or WIBD.
+    :param output_dir: The directory where converted NITF files will be saved.
+                       If not provided, a default 'output' directory will be created
+                       inside the input directory.
+    :raises Exception: If an invalid target format is provided.
+    """
+    if output_dir is None:
+        output_dir = os.path.join(input_dir, "output")
+
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    input_files = glob.glob(os.path.join(input_dir, "*.tif"))
+
+    for input_file in input_files:
+        output_file = os.path.join(output_dir, os.path.splitext(os.path.basename(input_file))[0] + ".ntf")
+        src_ds = gdal.Open(input_file)
+
+        if target_format == TargetFormat.GREYSCALE:
+            translate_to_greyscale(output_file, src_ds)
+        elif target_format == TargetFormat.WIBD:
+            translate_to_wibd(output_file, src_ds)
+        else:
+            raise Exception(f"Invalid target format: {target_format}")
+
+    print("Conversion completed.")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Convert GeoTIFF images to a target NITF format.")
+    parser.add_argument("input_dir", help="Input directory containing GeoTIFF images")
+    parser.add_argument("--output_format", help="Target format to convert to (options: GREYSCALE/WIBD)")
+    parser.add_argument("--output_dir", help="Output directory for NITF images (default: input_dir/output)", type=str)
+    parser.add_argument(
+        "--include_xml", help="Output directory for NITF images (default: NO, options: NO/YES)", type=str, default="NO"
+    )
+    parser.add_argument("--compression", help="Output directory for NITF images (default: LZW)", type=str, default="LZW")
```

Comment:
NIT: Cut and paste error in the description of the include_xml and compression arguments.